test/multicall/Multicall.js

```diff
@@ -0,0 +1,319 @@
+            data.push(encodeLongInput(1, 2, 3, 4, method));
+            addresses.push(reverter.address);
+        }
+
+        const callData = addresses.map( (addr, i) => {
+            return {
+                target: addr,
+                callData: data[i]
+            }
+        });
+
+        return {callData, data};
+    }
+
+    function encodeLongOutput(n, method) {
+        return solidityPack(
+            ["bytes4", "bytes"],
+            [
+                receiver.interface.getSighash(`${method}(uint24)`),
+                defaultAbiCoder.encode(["uint24"], [n]),
+            ]
+        );
+    }
+
+    async function multicallProcessLongInput(callData, overrides = {}) {
+        return await multicaller.callStatic.aggregate3(callData, {
+            gasLimit: 15_000_000,
+            ...overrides
+        });
+    }
+
+    async function multiDelegateCallProcessLongInput(callData, overrides = {}) {
+        return await multicaller.aggregate3(callData, {
+            gasLimit: 15_000_000,
+            ...overrides
+        });
+    }
+
+    async function multicallProcessLongOutput(n) {
+        const callData = [];
+        for (let i = 0; i < n; i++) {
+            callData.push({
+                callData: encodeLongOutput(n, 'processLongOutput'),
+                target: receiver.address
+            });
+        }
+
+        return multicaller.callStatic.aggregate3(callData, {
+            gasLimit: 15_000_000
+        });
+    }
+
+    async function multicallProcessLongOutputTx(n) {
+        const callData = [];
+        for (let i = 0; i < n; i++) {
+            callData.push({
+                callData: encodeLongOutput(n, 'processLongOutputTx'),
+                target: receiver.address
+            });
+        }
+
+        return multicaller.aggregate3(callData, {
+            gasLimit: 15_000_000
+        });
+    }
+
+    function getOutputLengthInBytes(res) {
+        let charSum = res.reduce(
+            (acc, value) => acc + value.length - 2, // do not count 0x
+            0
+        );
+
+        // 1 byte = 2 characters
+        return Math.floor(charSum / 2);
+    }
+
+    function getInputLengthInBytes(res) {
+        let charSum = res.reduce(
+            (acc, value) => acc + value.length,
+            0
+        );
+
+        // 1 byte = 1 character
+        return charSum;
+    }
+
+    before(async () => {
+        multicaller = await deployContract('Multicall3');
+        receiver = await deployContract('Receiver');
+        reverter = await deployContract('Reverter');
+    });
+
+    describe('static calls with large input', async function () {
+        it('should be able to make processLongInput calls with length 10', async function () {
+            const n = 10;
+            const {callData, data} = prepareLongInputData(n, 'processLongInput');
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(2660);    // data is 2.6 kb
```

Comment:
I see from here and subsequent tests that 266 is the length of the input data single element - not sure why.  Is that some kind of standard?  Computed in this test somewhere?  Use named constants here perhaps, for 10 * INPUT_ELEMENT_LENGTH or something like that?